src/phases/game-over-phase.ts

```diff
@@ -60,6 +60,11 @@ export class GameOverPhase extends BattlePhase {
           this.scene.ui.fadeOut(1250).then(() => {
             this.scene.reset();
             this.scene.clearPhaseQueue();
+            // If this is a ME, clear any residual visual sprites before reloading
+            const encounter = this.scene.currentBattle.mysteryEncounter;
+            if (encounter && encounter.introVisuals) {
```

Comment:
NIT:
Could be simplified right?
```suggestion
            if (encounter?.introVisuals) {
```